Signup: Clean up the P2 flow

southp opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to Automattic/martech#3708

## Proposed Changes

This PR fully removes the P2 signup flow.

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

We no longer actively sell P2 as a product. The landing page, https://wordpress.com/p2, also doesn't link to this flow anymore. It's time to clean up the code.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Cherry-pick a few signup flows powered by the classic framework, e.g. /start/domains, /start/personal, /start/business, and they should work as expected.
* Log in from `/log-in` should work as expected.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

southp committed:
Remove P2 signup flow steps

The P2 signup flow has been deprecated and replaced with a new flow.
This commit removes the old P2-specific signup steps that are no longer used.

southp committed:
Remove P2 signup flow and related components

This commit removes the P2 signup flow and its related components that are no
longer needed. The cleanup includes:

- Removing P2 signup form components and styles
- Removing P2 processing screen and wrapper components
- Cleaning up P2-related code from signup flows
- Removing unused P2 imports and references

This simplifies the codebase by removing deprecated functionality.

southp committed:
Remove the p2 flow specific code from the general purpose signup component

southp committed:
Undo unncessary edits done by Cursor

southp committed:
Remove dangling code from the already removed P2 flow checking block

southp committed:
Undo an unnecessary edit done by the Cursor agent

southp committed:
Add back a new line.

southp committed:
Remove the P2 flow step definitions

southp committed:
Exclude `user` from "there should be no unused steps" test.

tyxla commented on client/blocks/signup-form/index.jsx:
```diff
@@ -1387,8 +1330,6 @@ export default connect(
 			isWoo: getIsWoo( state ),
 			isWooJPC,
-			isP2Flow:
-				isP2Flow( props.flowName ) || get( getCurrentQueryArguments( state ), 'from' ) === 'p2',
```
In the codebase there are a bunch more checks where we check the if `from` query argument is `p2`. Let's clean those up too!

Here's one example:

https://github.com/Automattic/wp-calypso/blob/7236a424def0788ca8e64915c3b9c1097f802777/client/login/controller.js#L59

southp replied:
Nice find! Cleaned by https://github.com/Automattic/wp-calypso/pull/101712/commits/fd2fdec3a50e90fb4f45620c7b3494b6c94d56c0, https://github.com/Automattic/wp-calypso/pull/101712/commits/994f242ecc36d5941097a9d398f6ddb8f38319de and https://github.com/Automattic/wp-calypso/pull/101712/commits/fba00de7b3b12b1725c918b6f2ea0091ec050399.

tyxla commented on client/blocks/signup-form/p2.scss:
```diff
@@ -1,133 +0,0 @@
-@import "calypso/assets/stylesheets/p2-extends";
-
-.is-p2-invite.invite-accept,
-.is-p2-signup .signup__step.is-user {
```
There are a bunch of unused `.is-p2-signup` styles still in the codebase. In addition to removing them, we should be able to remove the `addP2SignupClassName` and `removeP2SignupClassName` helpers.

southp replied:
Done in [32973a2](https://github.com/Automattic/wp-calypso/pull/101712/commits/32973a2591ee9bf01ffeb31e6a612882237c2056)

tyxla commented on client/blocks/signup-form/p2.scss:
```diff
@@ -1,133 +0,0 @@
-@import "calypso/assets/stylesheets/p2-extends";
-
-.is-p2-invite.invite-accept,
```
Is the p2-specific invite flow another area that we can clean up? I know we have a dedicated set of component specifically for handling that flow for p2s.

southp replied:
Good point. Although my intuition is that we can, I didn't have a chance to study the p2 invite flow yet. Thus I'd like to skip cleaning them up for now.

tyxla replied:
Agreed. Sounds good as a follow up task 👍 Although, see my other comments

tyxla commented on client/signup/config/flows-pure.js:
```diff
@@ -13,28 +13,6 @@ const noop = () => {};
 	isEnabled( 'signup/social-first' ) ? 'user-social' : 'user';
 
-const getP2Flows = () => {
-	return isEnabled( 'p2-enabled' )
```
This `p2-enabled` feature flag can actually be removed from a few more files as well.

southp replied:
Cleaned a few more in [12326b6](https://github.com/Automattic/wp-calypso/pull/101712/commits/12326b640a7abd39c777f8dd7af5ae3c1a617e58).

tyxla commented on client/signup/config/steps-pure.js:
```diff
@@ -26,7 +26,6 @@ export function generateSteps( {
 	addAddOnsToCart = noop,
 	createAccount = noop,
 	createSite = noop,
-	createWpForTeamsSite = noop,
```
This one is still used:

https://github.com/Automattic/wp-calypso/blob/4afcedf36ad7a858f61ac5d087015ca9471af278/client/signup/config/steps.js#L8

Let's remove that usage and `createWpForTeamsSite` altogether.

southp replied:
Removed it in e36f4f58785

tyxla commented on client/signup/config/steps-pure.js:
```diff
@@ -39,8 +38,6 @@ export function generateSteps( {
 	isDomainFulfilled = noop,
 	maybeRemoveStepForUserlessCheckout = noop,
 	createSiteAndAddDIFMToCart = noop,
-	excludeStepIfEmailVerified = noop,
-	excludeStepIfProfileComplete = noop,
```
Can we also remove these completely? They don't seem to be used anymore.

southp replied:
Good point. I've cleaned them up in 58d8d1b6509. They look quite p2-flow specific.

tyxla commented on client/signup/is-flow.ts:
```diff
@@ -1,7 +1,4 @@
-export const isP2Flow = ( flowName: string ) => {
-	return flowName === 'p2';
```
There are similar checks in the codebase that don't use this helper, let's clean them up. One example:

https://github.com/Automattic/wp-calypso/blob/42613bf497329dfc20989fa962ec2cc65e5dfda3/client/lib/signup/step-actions/index.js#L1178

southp replied:
The provided instance has been removed by 58d8d1b6509. I don't see other instances from a grep. If I'm missing anything, please surely let me know :)

tyxla commented on client/signup/main.jsx:
```diff
@@ -104,22 +101,6 @@ function dependenciesContainCartItem( dependencies ) {
-		return;
-	}
-
-	document.body.classList.add( 'has-loading-screen-signup' );
```
There are existing styles for this class that we can remove.

southp replied:
After several more clean-up commits, I can't find this class is in-use anymore. If I'm missing anything, please feel free to let me know :)

tyxla commented on client/signup/p2-processing-screen/index.jsx:
```diff
@@ -1,51 +0,0 @@
-	return (
-		<div className="p2-processing-screen">
-			<div className="p2-processing-screen__logo">
-				<img src="/calypso/images/p2/logo-white.png" width="67" height="32" alt="P2 logo" />
```
Let's remove these images if they're not used. There are a few of them in this file.

southp replied:
Removed in 5ce9df52322

tyxla commented on client/signup/p2-step-wrapper/style.scss:
```diff
@@ -1,134 +0,0 @@
-@import url( https://fonts.googleapis.com/css2?family=Inter:wght@400;500;600;900&display=swap );
-@import "@wordpress/base-styles/breakpoints";
-@import "@wordpress/base-styles/mixins";
-@import "calypso/assets/stylesheets/p2-extends";
```
Let's try removing all those p2 extends. Some of them might still be used in login and invites, but I'd debate if that's necessary.

southp replied:
Good point. While I think we should keep P2 login and P2 invites for now, I don't think removing the p2 extends would make them terribly broken. Removed in aaeb09d513c.

tyxla commented on client/signup/steps/p2-site/index.jsx:
```diff
@@ -1,637 +0,0 @@
-import { login } from 'calypso/lib/paths';
-import wpcom from 'calypso/lib/wp';
-import P2StepWrapper from 'calypso/signup/p2-step-wrapper';
-import { getValueFromProgressStore } from 'calypso/signup/utils';
```
We should be able to remove the `getValueFromProgressStore` utility now, too.

southp replied:
Done in 6146bfca079

tyxla reviewed (commented):
Thanks for working on it @southp! Great to see so much red! (cc @ciampo for sharing the joy)

I see some more untapped opportunities for cleanup, and left some inline suggestions to propose them.

Let me know what you think!

southp committed:
Remove isP2Login from several login components

southp committed:
Remove `isP2Login` from LoginBlock

southp committed:
Remove `isP2Login` from LoginForm

southp committed:
Clean up `is-p2-signup`

southp committed:
Remove createWpForTeamsSite in full

southp committed:
Clean up the unused `excludeStepIfEmailVerified` and
`excludeStepIfProfileComplete`

southp committed:
Remove `getValueFromProgressStore`

southp committed:
Remove the unused P2 flow assets

southp commented:
@tyxla Thanks for the thorough review. I'm impressed by how you managed to find those places worth further cleaning up :) However, from my testing it looks like I have introduced some regression that I'm yet to find out at the moment, so I've labeld this PR as In Progress now. I will let you know when it's ready again 🙇🏼

southp committed:
Remove p2 extended style references

tyxla commented:
Thanks for the great work on it @southp 🙌 Over 3K lines of red is quite impressive already 🔴 !

Let me know once it's ready for another look.

southp commented:
@tyxla It turned out to be a false alarm. The layout is expected to be a bit different at the moment since [the step container v2 is enabled for the development environment](https://github.com/Automattic/wp-calypso/blob/trunk/config/development.json#L142):
![CleanShot 2025-03-25 at 10 15 51@2x](https://github.com/user-attachments/assets/8c2949cd-37e3-4843-a288-2c363c4b069d)
Although `test/e2e/specs/onboarding/onboarding__write.ts` is failing like 90% of time for me, I don't think it relates since I can reproduce the failing on trunk as well. Thus I'm opening this PR up for review again and will investigate that e2e test failure in parallel.

tyxla commented on client/blocks/login/index.jsx:
```diff
@@ -352,7 +352,6 @@ class Login extends Component {
 			isGravPoweredLoginPage,
 			isJetpack,
 			isManualRenewalImmediateLoginAttempt,
-			isP2Login,
```
This prop appears in the `LayoutLoggedOut` component, too. 

The good news? There's more related CSS we can remove, because removing this usage will also remove the last use of the `is-p2-login` classname, and there are still (a bunch of!) unused styles declared under that classname.

The bad news? I think removing it should be part of cleaning up p2 invites, since, theoretically, one can still land on the logged-out page coming from an old p2 invitation:

https://github.com/Automattic/wp-calypso/blob/a78ca19a71272ffc9027828c144da2b3d9691d79/client/my-sites/invites/invite-accept-logged-out/index.jsx#L47

So, not something to try in this already large PR. This could make a great follow-up, though.

See my other comments - TL;DR I think we should clean this up here, in this PR.

southp replied:
`is-p2-login` in `LayoutLoggedOut` usage and its dangling CSS rules are removed by https://github.com/Automattic/wp-calypso/pull/101712/commits/d479c5e03d8ab72dae699912ffcfed251064017f. I will follow up the p2-invite things in the separate comment below.

tyxla commented on client/blocks/signup-form/index.jsx:
```diff
@@ -1060,19 +1038,6 @@ class SignupForm extends Component {
 
-		if ( this.props.isP2Flow ) {
-			return (
-				<div className="signup-form__p2-footer-link">
```
I spotted we're still using this `classname` for the Blaze Pro signup, but it doesn't have any associated styles. Looks like a copy-pasta issue. Should be safe to remove.

southp replied:
Ah, good finding. Removed by [a9fdb17](https://github.com/Automattic/wp-calypso/pull/101712/commits/a9fdb178f0e276c6e9f6c38065df30b835506dfd)

tyxla commented on client/login/wp-login/index.jsx:
```diff
@@ -400,13 +398,6 @@ export class Login extends Component {
 			return null;
 		}
 
-		if ( isP2Login && currentQuery?.redirect_to ) {
-			const urlParts = getUrlParts( currentQuery.redirect_to );
-			if ( urlParts.pathname.startsWith( '/accept-invite/' ) ) {
-				return null;
-			}
-		}
```
Hmm, we'll still need to test p2 invitation flows thoroughly before shipping. This seems like part of that flow.

tyxla replied:
I tested it myself by doing this:

- Using one of my old p2 sites
- Sending an invitation to one of my test accounts
- Starting as a logged-out user
- Clicking the link in the email to accept the invite
- This is what I saw, I think it is acceptable (we can live with som login link duplications IMHO):

![Screenshot 2025-03-25 at 9 48 06](https://github.com/user-attachments/assets/0c215b88-5ad6-424e-9135-5d741f6f44f3)

- Then, clicking the "Log in" link leads me to a regular login form:

![Screenshot 2025-03-25 at 9 49 08](https://github.com/user-attachments/assets/705a16b0-0372-42dc-a6a2-6e2d2bfb602b)

- I compared the login form to what we see on staging, and I don't see any differences.

Then, this is what I saw after logging in:

![Screenshot 2025-03-25 at 9 53 33](https://github.com/user-attachments/assets/8f64eaca-9af7-4332-a2f8-eb59345c182e)

Doesn't feel perfect, but given that this is a rare flow nowadays, I think it's OK as-is.

FWIW, here's how it looks on staging:

![Screenshot 2025-03-25 at 9 57 49](https://github.com/user-attachments/assets/9c87b0c1-dc3f-41d4-adf1-eba06e675e58)

Based on the above, I think we're mostly good, but we may need to do some minor improvements.

I would still recommend cleaning up the rest of the p2-specific styles for invites, if possible in this PR, since we're already affecting them. Ideally things should be as simple as possible, enough to not be broken.

southp replied:
You are right. I attempted to identify the minimal required pieces for the p2 invite flow to be working by creating a p2 site for my test account via `https://wordpress.com/start/p2` and then sending the invitation to my other test accounts. [3fa7dc5](https://github.com/Automattic/wp-calypso/pull/101712/commits/3fa7dc5c713f432eadb5c2df6cb0ae8032c1528e) is what I've got –– accepting the invitation both as a logged-in account and logged-out doesn't look terribly broken:

| Logged-out | Logged-in|
|------------|------------|
| ![CleanShot 2025-03-26 at 16 45 30@2x](https://github.com/user-attachments/assets/c538a8f3-3da8-4a06-a6ce-5bc59f3766b0) | ![CleanShot 2025-03-26 at 16 44 30@2x](https://github.com/user-attachments/assets/0c2f10d2-f6d9-47a9-abbc-25009e04fb63) |

Removing the rules in `client/my-sites/invites/p2/style.scss` would make it terribly broken, so I choose to keep them for now:

| Ouch |
|-------|
| ![CleanShot 2025-03-26 at 16 56 54@2x](https://github.com/user-attachments/assets/1ee81515-e335-4995-acf7-ce6ed49ad26b) |

Does that make sense to you?

tyxla replied:
Absolutely. Thanks for taking the time to address it thoroughly 👍

tyxla commented on client/login/wp-login/index.jsx:
```diff
@@ -624,7 +610,6 @@ export class Login extends Component {
 				</Main>
 
 				{ this.renderFooter() }
-				{ this.props.isP2Login && this.renderP2PoweredBy() }
```
`renderP2PoweredBy()` can be removed now, together with its styles.

southp replied:
Also removed by [563f699](https://github.com/Automattic/wp-calypso/pull/101712/commits/563f6995733d14f5b04b9b08bb226b45dae8bce5)

tyxla commented on client/login/wp-login/index.jsx:
```diff
@@ -596,7 +583,6 @@ export class Login extends Component {
 
 		return (
 			<div>
-				{ this.props.isP2Login && this.renderP2Logo() }
```
`renderP2Logo()` can be removed now as well, together with its styles.

southp replied:
Removed by [563f699](https://github.com/Automattic/wp-calypso/pull/101712/commits/563f6995733d14f5b04b9b08bb226b45dae8bce5)

tyxla reviewed (commented):
Thanks for the great work @southp 🙌 

I think we're nearly there! I'd recommend doing the invites cleanup, since we're already affecting that flow and we might want to apply minor polish after fully cleaning up, just so it doesn't look broken. See inline comments for more details.

southp committed:
Remove `renderP2Logo`, `renderP2PoweredBy` helper functions, their
accompanying styles, and images/p2/logo.png

southp committed:
Remove `p2-extends` stylesheet

southp committed:
Remove the dangling class `signup-form__p2-footer-link`

southp committed:
Remove `isP2Login` from LayoutLoggedOut

southp committed:
Clean up the dangling CSS rules for `is-p2-signup`

southp committed:
Remove the styling rules from the P2 invite flow as much as possible

tyxla commented:
@southp I'll need to do some final testing and review, and I'm not sure if I'll be able to get to it today, but in any case, this is an amazing cleanup. Thank you 🙌

southp commented:
Sure thing, @tyxla :) Thanks for bearing with me; it's impossible to reach this form without your help. I wanted to find 🙇🏼 emoji reaction for your comment but it's not there –– github should add it with a major PR campaign 🤣

tyxla reviewed (approved):
Thanks once again for the amazing work on this one @southp 🙌 ❤️ 

The p2 invitation form tests well, and what we've kept is a good compromise IMHO.

Gave the PR a thorough run and I couldn't spot any other issues. Also, I believe we've removed everything we could, related to these p2 flows.

Feel free to ship after rebasing to resolve the config conflicts 🚀

southp committed:
Remove P2 flow and clean up related config flags

Remove the P2 flow and clean up related config flags across all environments.
This includes removing the p2/p2-plus feature flag from all config files and
cleaning up the associated flow in flows-pure.js.

southp committed:
Remove the remnants of `p2-enabled`

southp committed:
Remove two redundant feature flags from a wrong conflict merge

southp committed:
Move the trail map feature flag back to the original place.
